### PR TITLE
Check that generated docs and mocks are up-to-date

### PR DESCRIPTION
**Pull request by gkrenn**
## Description

Our generated markdown files and mockery mocks are sometimes outdated. 
This PR adds a CI step to check if those files were updated.

In addition it: 
- removes an old check for the docs that did not work 
- update the `make doc` scripts to actually work in the CI 
- add an `issue-845-fix: True` annotation to the mockery config as the edge case anyway does not apply to our codebase
- update all outdated files

[Jira](https://dt-rnd.atlassian.net/browse/DAQ-1463)

## How can this be tested?

- Check if CI ran through
- Change one `md` file and run `act -j generated-files` locally => it should fail

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -113,11 +113,27 @@ jobs:
...
         id: deepcopy
         run: |
           make manifests/deepcopy
           git diff --exit-code
+      - name: Check doc
+        id: doc
+        run: |
+          make doc
+          git diff --exit-code
+      - name: Mockery
```
Comment (andriisoldatenko) [inline .github/workflows/ci.yaml:131]:
```suggestion
      - name: Check no changes for mocks
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -113,11 +113,27 @@ jobs:
       - uses: actions/setup-go@41dfa10bad2bb2ae585af6ee5bb4d7d973ad74ed # v5.1.0
         with:
           go-version-file: "${{ github.workspace }}/go.mod"
+      - name: Set up Helm
```
Comment (andriisoldatenko) [inline .github/workflows/ci.yaml:116]: why do we need this?

**Comment by andriisoldatenko** [comment]
left small renaming suggestions, otherwise LGTM!

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (gkrenn) [reply .github/workflows/ci.yaml:116]: `make doc` depends on `make manifests` which uses helm to generate the files

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (gkrenn) [reply .github/workflows/ci.yaml]: Renamed

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -113,11 +113,27 @@ jobs:
...
+        uses: azure/setup-helm@fe7b79cd5ee1e45176fcad797de68ecaf3ca4814 # v4.2.0
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: actions/setup-node@39370e3970a6d050c480ffad4ff0ed4d3fdee5af # v4.1.0
       - name: Check deepcopy files
         id: deepcopy
         run: |
           make manifests/deepcopy
           git diff --exit-code
+      - name: Check doc
```
[the remark this answers is not in the document]
Comment (gkrenn) [reply .github/workflows/ci.yaml]: Renamed

**Review by andriisoldatenko — APPROVED** [review]

**Review by waodim — APPROVED** [review]
lgtm

**Review by albertogdd — APPROVED** [review]

**Pull request merged**